### DISCUSSION TITLE
Build Errors Related to @material-ui/core

### DISCUSSION THREAD

**Discussion by JennaSys** [rtp-book/project#4] [Errata]
If you are experiencing problems getting the project to build with errors related to `@material-ui/core`, try changing the dependency in the [package.json](https://github.com/rtp-book/project/blob/main/client/package.json) file to remove the caret (^) from the version number for that library so that npm doesn't install a later version:

`"@material-ui/core": "4.11.0"`

There was apparently a breaking change in version 4.12.0 in the `styles` module that would otherwise require refactoring of the project source code.

**Comment by JennaSys** [comment]
I originally put forth pinning the Material-UI version as a fix for this issue.  But as another workaround that addresses the issue directly, @logc let me know about a code change you can make to the import of `CreateMuiTheme` that will also work. This fix actually addresses the core problem where `CreateMuiTheme` has been renamed to `CreateTheme` in Material-UI versions after 4.11.0.

In the [_client/src/common/pymui.py_](https://github.com/rtp-book/project/blob/ad3a563ce802bef87629f1ab77a0f299050986d7/client/src/common/pymui.py#L49) module, change the import of `CreateMuiTheme` from this:
```python
createMuiTheme = require('@material-ui/core/styles/createMuiTheme')['default']
```
to this:
```python
createMuiTheme = require('@material-ui/core/styles/createTheme')['default']
```
So instead of locking in the version, this approach also works without having to change any other source code.

&nbsp;  

&nbsp;  

_Side Note..._
Interestingly enough, the original `createMuiTheme` name still exists and can even be imported by making this change:
```python
createMuiTheme = require('@material-ui/core/styles/createTheme')['createMuiTheme']
```
which allows the source code to compile properly, but note that this will result in a deprecation warning about the name change when you run it:
```bash
Material-UI: the createMuiTheme function was renamed to createTheme.
You should use `import { createTheme } from '@material-ui/core/styles'`
```

**Reply by logc** [reply]
One comment about the side note: as far as I understand, the function is deprecated and will be removed in some later version 5, possibly also on version 4. You can still import it (and ignore the deprecation notice) in 4.12, but it may disappear in 4.x later on.

**Reply by JennaSys** [reply]
Yep, this is definitely not a good thing to do. With any library, using "undocumented" features is risky that way.

**Comment by dvshmkr** [comment]
Finishing chapter 19 now.  I'm was able to get it to work as described by running the  flask server as a separate process from the parcel server. Just want to make sure that is correct/expected.

**Reply by JennaSys** [reply]
Yes, in this case the Flask server is run as a separate process from the parcel development server.

In a production environment, the Flask server would (or at least could) also serve up the generated React JavaScript/html as static files as well.
